Frontend docs for new dark mode themes (#17960)
home-assistant/home-assistant.github.io: source/_integrations/frontend.markdown

## Defining Themes

> ### Theme format
> The frontend integration allows you to create custom themes to influence the look and feel of the user interface.

Reviewer: ```suggestion
### Theme format

The frontend integration allows you to create custom themes to influence the look and feel of the user interface.
```

      text-primary-color: purple
      mdc-theme-primary: plum
    sad:
      primary-color: steelblue
```

> The example above defines two themes named `happy` and `sad`. For each theme you can set values for CSS variables. If you want to provide hex color values, wrap those in apostrophes, since otherwise YAML would consider them to be comments (`primary-color: '#123456'`). For a partial list of variables used by the main frontend see [ha-style.ts](https://github.com/home-assistant/home-assistant-polymer/blob/master/src/resources/ha-style.ts).

Reviewer: ```suggestion
The example above defines two themes named `happy` and `sad`. For each theme, you can set values for CSS variables. If you want to provide hex color values, wrap those in apostrophes, since otherwise, YAML would consider them a comment (`primary-color: '#123456'`). For a partial list of variables used by the main frontend see [ha-style.ts](https://github.com/home-assistant/home-assistant-polymer/blob/master/src/resources/ha-style.ts).
```

> ### Dark mode support
> Starting with version 2021.6 it is also possible to create themes that are based on the default dark mode theme. New themes can also support both light and dark mode and allow the user to switch between those on the user profile page:

Reviewer: ```suggestion
### Dark mode support

It is also possible to create themes that are based on the default dark mode theme. New themes can also support both light and dark mode and allow the user to switch between those on the user profile page:
```

> [![Open your Home Assistant instance and show your Home Assistant user's profile.](https://my.home-assistant.io/badges/profile.svg)](https://my.home-assistant.io/redirect/profile/)

Reviewer: ```suggestion
{% my profile badge %}
```

Extended example to show the mode definitions.

```yaml
# Example configuration.yaml entry
frontend:
  themes:
    happy:
      primary-color: pink
      text-primary-color: purple
      mdc-theme-primary: plum
    sad:
      primary-color: steelblue
      modes:
        dark:
          secondary-text-color: slategray
    day_and_night:
      primary-color: coral
      modes:
        light:
          secondary-text-color: olive
        dark:  
          secondary-text-color: slategray
```

Theme `happy`: Same as in the previous example. This legacy format is still supported and will behave as before and automatically use the default light theme as the base.

> Theme `sad`: By using the new `mode` key plus the sub key `dark` this theme will now be based on the default dark theme. The final theme rules are determined in three steps: First the default dark theme CSS variables will be applied, then second the CSS variables from the top level of the theme that are mode-independent (`primary-color: steelblue` in this example) and lastly the mode specific CSS variables will be layered on top (`secondary-text-color: slategray`).

Reviewer: ```suggestion
Theme `sad`: By using the new `mode` key plus the subkey `dark` this theme will now be based on the default dark theme. The final theme rules are determined in three steps: First, the default dark theme CSS variables will be applied, then second the CSS variables from the top level of the theme that are mode-independent (`primary-color: steelblue` in this example) and lastly the mode-specific CSS variables will be layered on top (`secondary-text-color: slategray`).
```
